lib/runner.js

```diff
@@ -202,10 +203,13 @@ Runner.prototype.fail = function(test, err) {
   ++this.failures;
   test.state = 'failed';
 
-  if (!(err instanceof Error)) {
+  var isError = err instanceof Error;
+  var isErrorLike = err && isString(err.message) && isString(err.name);
```

Comment:
this will throw an exception if err happens to be `null`.  add additional checks.  

I would also probably avoid creating the variables if we are not going to use them elsewhere.